runtime/near-vm-logic/src/mocks/mock_external.rs

```diff
@@ -209,6 +248,46 @@ impl External for MockedExternal {
     fn validator_total_stake(&self) -> Result<Balance> {
         Ok(self.validators.values().sum())
     }
+
+    #[cfg(feature = "protocol_feature_function_call_weight")]
+    fn distribute_unused_gas(&mut self, gas: Gas) -> GasDistribution {
+        let gas_weight_sum: u128 =
+            self.gas_weights.iter().map(|(_, GasWeight(weight))| *weight as u128).sum();
+        if gas_weight_sum != 0 {
+            let gas_per_weight = (gas as u128 / gas_weight_sum) as u64;
```

Comment:
if someone sets very large 'gas_weight' - then gas_per_weight can be 0, right? 
(we should mention it in the comment)

Reply:
Added docs in vm logic for this, let me know if it's preferable to duplicate docs to the external trait and/or impls

Reply:
Thanks for adding those. I'd also add a one-line comment here, that we're using a floor, and we know that it might be 0 if the sum is too large.